test/Datadog.Trace.ClrProfiler.IntegrationTests/SmokeTests/SmokeTestBase.cs

```diff
@@ -131,5 +147,48 @@ protected void CheckForSmoke(bool shouldDeserializeTraces = true)
             Assert.True(successCode == result.ExitCode, $"Non-success exit code {result.ExitCode}");
             Assert.True(string.IsNullOrEmpty(result.StandardError), $"Expected no errors in smoke test: {result.StandardError}");
         }
+
+        private class OutputReader
+        {
+            private readonly StreamReader _reader;
+            private readonly StringBuilder _buffer = new StringBuilder();
+            private readonly CancellationToken _cancellationToken;
+            private readonly Thread _thread;
+
+            public OutputReader(StreamReader reader, CancellationToken token)
+            {
+                _reader = reader;
+                _cancellationToken = token;
+
+                _thread = new Thread(Drain) { IsBackground = true };
+                _thread.Start();
+            }
+
+            public string GetOutput(bool waitForCompletion = false)
+            {
+                if (waitForCompletion)
+                {
+                    _thread.Join();
+                }
+
+                lock (_buffer)
+                {
+                    return _buffer.ToString();
+                }
+            }
+
+            private void Drain()
+            {
+                while (!_reader.EndOfStream && !_cancellationToken.IsCancellationRequested)
+                {
+                    var line = _reader.ReadLine();
+
+                    lock (_buffer)
+                    {
+                        _buffer.Append(line);
```

Comment:
Should we use `AppendLine()` since the result from `ReadLine()` does not include the EOL?

Reply:
Yeah, I only realized that after I had already approved the PR. Yeah, we should follow up and modify that to `AppendLine()`